### PR TITLE
Add a RESTful API for external configuration

### PR DESCRIPTION
**Pull request by raphaelm**
This PR adds a RESTful API to manage world configuration and rooms. This is **not** intended to be used by the venueless webapp, I still think we should do as much websocket as possible there.

Instead, this is intended to be used by external systems like pretix or pretalx to automatically push over room configurations etc.

For simplicity, I opted to use the very same JWT configuration for authentication with a few extra permission flags, even though the API authentication currently does not persist users to the venueless database (no need to).

Any changes to the room or world config result in the world configuration to be pushed to all connected clients.

If you agree with the general design, I would add documentation as well.

TODO:

- [x] Auto-create channels whenever a chat module is added to a room
- [x] Docs

### CODE REVIEW

File: server/sample/worlds/sample.json [hunk 1]
```diff
@@ -24,10 +24,13 @@
     ]
   },
   "permissions": {
+    "world.api": ["admin", "api"],
+    "world.secrets": ["admin", "api"],
     "world.update": ["admin"],
     "world.announce": ["admin"],
     "room.create": ["admin"],
     "room.announce": ["admin"],
+    "room.inspect": ["admin"],
```
Comment (rixx) [inline server/sample/worlds/sample.json:33]: What does ``room.inspect`` do?

File: server/venueless/api/auth.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import authentication, exceptions, permissions
+from rest_framework.authentication import get_authorization_header
+
+from venueless.core.models import Room, World
+from venueless.core.services.world import has_permission
+from venueless.core.utils.jwt import _decode_token
```
Comment (rixx) [inline server/venueless/api/auth.py:7]: We should probably rename the method to ``decode_token`` if we're using it in a different module.

File: server/venueless/api/auth.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            perm_key = "room.delete"
+        if perm_key:
+            return perm_key in permset and has_permission(
+                permset[perm_key], request.auth.get("traits")
+            )
+        else:
+            return True
+
+
+class AnonymousUser:
```
Comment (rixx) [inline server/venueless/api/auth.py:112]: Huh, why is this necessary?

File: server/venueless/live/modules/auth.py [hunk 4]
```diff
@@ -44,6 +41,9 @@ async def login(self):
         await self.consumer.channel_layer.group_add(
             f"user.{self.consumer.user['id']}", self.consumer.channel_name
         )
+        await self.consumer.channel_layer.group_add(
+            f"world.{self.world}", self.consumer.channel_name
```
Comment (rixx) [inline server/venueless/live/modules/auth.py:45]: I wonder if we can find a way to document the groups we use, because this is starting to take up my working memory. Either a list somewhere or a data abstraction? Or just depend on the fact that the list won't grow that much and that we can remember what we are using?

File: server/venueless/settings.py [hunk 5]
```diff
@@ -136,22 +135,24 @@
...
     CSRF_COOKIE_DOMAIN = os.getenv("VENUELESS_COOKIE_DOMAIN", "")
 
-SESSION_COOKIE_SECURE = (
-    os.getenv("VENUELESS_HTTPS", "True" if SITE_URL.startswith("https:") else "False")
-    == "True"
-)
+# SESSION_COOKIE_SECURE = (
+#     os.getenv("VENUELESS_HTTPS", "True" if SITE_URL.startswith("https:") else "False")
+#     == "True"
+# )
```
Comment (rixx) [inline server/venueless/settings.py:144]: You can probably just delete all the cookie settings right away, I don't think we'll need them any time soon?

File: server/sample/worlds/sample.json (hunk 1 above)
Comment (raphaelm) [reply server/sample/worlds/sample.json:33]: I'll double-check, probably nothing after I added world.secrets. The idea was something like "can see room configuration" as opposed to a possible "room.view" (attendee can see that room exists)

File: server/venueless/api/auth.py (hunk 2 above)
Comment (raphaelm) [reply server/venueless/api/auth.py:7]: Yeah, and what about the thing that currently is _decode_token? I was trying to avoid to many sync_to_async and async_to_sync's in one view, as that sounds bad from a performance point of view

File: server/venueless/api/auth.py (hunk 3 above)
Comment (raphaelm) [reply server/venueless/api/auth.py:112]: something something rest framework exception. Otherwise, DRF tries to import django.contrib.auth.AnonymousUser and fails since django.contrib.auth isn't set up

File: server/venueless/live/modules/auth.py (hunk 4 above)
Comment (raphaelm) [reply server/venueless/live/modules/auth.py:45]: We should at least list it, yes!

File: server/venueless/settings.py (hunk 5 above)
Comment (raphaelm) [reply server/venueless/settings.py:144]: Probably, yes

File: server/venueless/api/auth.py (hunk 2 above)
Comment (rixx) [reply server/venueless/api/auth.py:7]: How about making it a method on `World`? It comes out to the same thing and is more  understandable, I'd think. `world.decode_token()` seems like a good API.
